### ISSUE TITLE
In Opera the onChange() is not triggered

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bdshadow/gwtupload#32]
```
What steps will reproduce the problem?
1. try http://gwtupload.alcala.org/gupld/
gwtuploadsample.SingleUploadSample/SingleUploadSample.html in Opera 10


What is the expected output? What do you see instead?
Upload won't work...
I am using the latest gwtupload version on Windows7


Seems like the onChange() handler in Opera is not called when selecting a 
file.  basename will be null and the onSubmit() will therefore be 
cancelled.

A workaround is to manually call the onChange() method in onSubmit() if 
basename is null...


```

Original issue reported on code.google.com by `bono...@gmail.com` on 14 Mar 2010 at 12:29

**Comment by GoogleCodeExporter** [comment]
```
Hello, it seems an opera issue. 
On Change event is fired but I don't understand when. 
For me, after selecting the file, and then clicking on the input again onChange 
is 
fired.
Could you investigate a bit more and check when onChange is fired in your 
version?
(I'm using opera 10 in linux). Do you know if there is a library able to handle 
this
behaviour?.

Thanks
Manolo
```

Original comment by `manuel.carrasco.m` on 15 Mar 2010 at 11:15
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
That's correct (after selecting the file, and then clicking on the input again 
onChange is fired).

It has something todo with the cache; if you clear the cache in Opera the 
trigger 
sequence is correct (like you would expect). After that, it is crewed again.
```

Original comment by `bono...@gmail.com` on 15 Mar 2010 at 11:49

**Comment by GoogleCodeExporter** [comment]
```
I forgot to mention that the onChange() isn't fired at all in Opera when the 
'cache 
is filled'
```

Original comment by `bono...@gmail.com` on 15 Mar 2010 at 11:51

**Comment by GoogleCodeExporter** [comment]
```
Definitively it seems opera fire the event when the focus changes. 
When the browse button is clicked the focus goes to it, and you have to push 
tab key 
or click over another focusable element in order to lose the focus and to fire 
the 
onChange event.
About the cache, I don't understand what you mean. Maybe, you are selecting the 
same 
file again, so the onChange is not called, or maybe Uploader is configured for 
avoiding to upload files which have been already sent, so the file is not 
uploaded. 
```

Original comment by `manuel.carrasco.m` on 15 Mar 2010 at 12:29

**Comment by GoogleCodeExporter** [comment]
```
Sorry about the confusion of the cache; it was my workaround that gave me the 
wrong 
idea...
```

Original comment by `bono...@gmail.com` on 15 Mar 2010 at 12:46

**Comment by GoogleCodeExporter** [comment]
```
Hello.

I've spotted where the problem is: gwt has a hack for avoiding firing the 
onChange 
event each time the user clicks on the file input box (when the user writes the 
file 
name instead of selecting it with the popup selector), this workaround is not 
needed 
in GWTUpload, so I've disabled it.
The new version (in beta status) fixes this and add a cool feature for 
customizing 
the choose button.
Please check if everything works in the project demo page or downloading the 
beta 
version.

Thanks
Manolo
```

Original comment by `manuel.carrasco.m` on 16 Mar 2010 at 5:42

**Comment by GoogleCodeExporter** [comment]
```
Hola,
Nice job! I' ve checked the demos in Opera, FF, Chrome, Safari (on windows) en 
IE8 
en *of course* only IE has problems..

In Single uploader, a javascript error is raised.
In Multiple uploaders, 1. the queue is cancelled when selecting another file 
and 2. 
the 'Choose File Dialog' opens more the one time when selecting multiple files.

Keep up the good work.
```

Original comment by `bono...@gmail.com` on 17 Mar 2010 at 10:33

**Comment by GoogleCodeExporter** [comment]
```
Hello

These IE8 issues should be fixed in version 0.5.8-M2. 
Please could you check if they happen?

Thx
Manolo
```

Original comment by `manuel.carrasco.m` on 5 Apr 2010 at 3:52

**Comment by GoogleCodeExporter** [comment]
Original comment by `manuel.carrasco.m` on 7 Apr 2010 at 2:13
- Changed state: **Fixed**
